### PR TITLE
Fix bug with gradient leaks at the GAN example.

### PR DESCRIPTION
**Pull request by kwanUm**
Following #591, the GAN example at the Pytorch Lightning project is leaking
gradients between the Generator and the Discriminator steps. This
fix clears the gradients of the Discriminator before returning its loss.
In the future, fix #603 may make this fix unnecessary.

This diff also adds a note for using SyncBatchNorm in case of training with DDP.

### CODE REVIEW

**Comment by williamFalcon** [comment]
@kwanUm is this still needed now that #603 is merged?

**Comment by williamFalcon** [comment]
@kwanUm closing this. If this needs to be re-opened let me know

**Comment by kwanUm** [comment]
Thanks, I'm concerned with the runtime speed of this new approach. Sent a
review comment at
https://github.com/PyTorchLightning/pytorch-lightning/pull/603/commits/67049457976c4b6e59b1ea3b275f2abcc9f64b61
